### PR TITLE
[CHIA-1976] bump chia_rs to 0.16.0 and introduce soft-fork6

### PR DESCRIPTION
**Pull request by arvidn**
### Purpose:

back port soft-fork 6 on top of the release tag

These are the changes to chia_rs: https://github.com/Chia-Network/chia_rs/compare/0.14.0...0.16.0

### New Behavior:

The new `chia_rs` implements the keccak softfork.

The activation heights in this PR are:

| network | block height |approx. date |
| --- | --- | --- |
| mainnet | 6800000 |2025-03-22 |
| testnet11 | 2000000 |2025-02-06 |

(please double check these calculations)

### CODE REVIEW

**Comment by aqk** [comment]
Hey, I noticed that you set 2025-02-28 as the approximate date to trigger on testnet11.

Wouldn it be better to set a date more like 2024-12-15 or 2025-01-05, so we can confirm the behaviour earlier?

File: chia/consensus/default_constants.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
     MAX_GENERATOR_SIZE=uint32(1000000),
     MAX_GENERATOR_REF_LIST_SIZE=uint32(512),  # Number of references allowed in the block generator ref list
     POOL_SUB_SLOT_ITERS=uint64(37600000000),  # iters limit * NUM_SPS
-    SOFT_FORK5_HEIGHT=uint32(5940000),
+    SOFT_FORK6_HEIGHT=uint32(6815000),
```
Comment (hoffmang9) [inline chia/consensus/default_constants.py:75]: If we want 124.8 days at the time of this comment, then this is correct.

**Review by hoffmang9 — DISMISSED** [review]

File: chia/consensus/default_constants.py (hunk 1 above)
Comment (arvidn) [reply chia/consensus/default_constants.py:75]: maybe that's overkill. I was thinking of a release on the 18th then + 30 days and round up. I see now cameron suggested a release on the 12th

Comment (arvidn) [reply chia/consensus/default_constants.py:75]: If I change it to 6800000 it's ~2025-03-22 which is closer to 2024-12-12 + 90

**Review by emlowe — APPROVED** [review]
Looks ok to me

File: chia/consensus/default_constants.py (hunk 1 above)
Comment (pmaslana) [reply chia/consensus/default_constants.py]: Resolving this conversation due to approval of the PR and changes.

**Pull request merged**